Adds craftable ash catchers

Xhuis opened the pull request:
:cl: Xhuis
add: You can now break down one goliath hide into three goliath hide chunks (and vice-versa) via the crafting menu.
add: Added craftable ash catchers from goliath hide chunks and metal rods (or bone). Ash catchers are portable tarps that can be erected in the wastes as both a primitive form of expansion for roles that don't have many materials to build with and to protect one from ash storms in the field. Mobs in a 3x3 area are immune to ash storms' burning effects while underneath the catcher.
add: Ash catchers can be cleared of ash by attacking them while on help intent. If not cleared periodically, the ash catcher may fall apart. Sprites will show if an ash catcher is at risk.
/:cl:


![Ash catchers](https://puu.sh/u95Lj/171cc079cf.png)

_Pictured: Showing off the sprites of the catchers and the changing alphas for mobs._

Ash catchers function based on a system centered around a variable called `ash_caught` that increases every tick the catcher is in an ash storm. At certain thresholds, the catcher's name, description, and sprite will change; these thresholds are 150 ticks, 300 ticks, and 400 ticks. At 400 ticks, the catcher will collapse from the weight and be destroyed.

People can activate the ash catcher with an empty hand on help intent to clear out the ash over five seconds and restore it to full integrity. It can also be torn down over seven seconds to be relocated, and takes seven more to set up again.

Catchers are made with one goliath hide chunk and two rods, or one goliath hide chunk and one bone. Goliath hide chunks are broken down into from one goliath hide, at a rate of 3 chunks/hide.

Commit:
Adds craftable ash catchers

RemieRichards commented:
Remove the alpha from the sprite, and have it be done programatically when someone walks under one (```Crossed()``` and ```Uncrossed()```)

RemieRichards commented on code/game/objects/structures/ash_catcher.dm:
```diff
@@ -0,0 +1,111 @@
+//If the catchers are too heavy and collect too much ash, they will collapse and allow ash storms through.
+/obj/structure/ash_catcher
+	name = "ash catcher"
+	desc = "A sheet of goliath's hide stretched over four metal rods."
```
it's also bone though.

Xhuis replied:
I kept it out of the description because it might be too spooky for people and I can't add a spook warning for everything.

(I'm gonna add a subtype in the next commit, don't worry!)

Xhuis commented:
Changing alphas could be neat. As it stands the tarp sprite has a low alpha but the poles don't. Any ideas of a reasonable alpha?

RemieRichards commented:
uh, 127? (half of full alpha), if it's too low, + 50, too high, - 50, rince and repeat with smaller numbers.

ChangelingRain commented on code/game/objects/structures/ash_catcher.dm:
```diff
@@ -0,0 +1,111 @@
+	desc = "A sheet of goliath's hide stretched over four metal rods."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "ash_catcher_intact"
+	layer = ABOVE_ALL_MOB_LAYER
```
ABOVE_MOB_LAYER, this goes above alien queens and such

Xhuis replied:
That was intentional but I can see how it would look silly.

KorPhaeron commented:
This is a pretty enormous waste of resources to protect 1 tile as far as base building goes,

Xhuis commented:
Recipes can use tweaking. I can just as easily change it to require like one bone.

KorPhaeron commented:
The only way they'd ever be viable for base building is if you got multiple per goliath hide.

I think these will only really see use as mobile emergency shelters that people drag around behind them, which will be kind of lame

Xhuis commented:
I was thinking about letting people break up the hide into "chunks", so I might do that if you're okay with it.

Anonmare commented:
Would be a little useful for non-miners on Lavaland.
You just need a welding tool, a bit of iron ore and goliath hide.

Commit:
Adaptive sprites and a cheaper recipe

Xhuis commented:
* Reduced recipe costs. Rod catchers now require two rods and a goliath hide chunk, while bone ones require one bone and one goliath hide chunk.
* Added goliath hide chunks. One piece of goliath hide can be broken down into three chunks in the crafting menu, or vice-versa.
* Ash catchers now have dynamic sprites when crossed! Catchers with mobs beneath them will be translucent, and normal ones will not. Note that this fully reveals the horrific quality of my sprites, so take care.
* Fixed clearing out actually adding ash.
* Allows ash catchers to work on any asteroid tile, not just basalt.
* Reduced setup time to 4 seconds down from 7.5.

Cheridan commented:
Why not just have the recipe make 3 shelters instead of having an awkward intermediate item

Xhuis commented:
I don't think that recipes support having multiple results, so I would have to have a dummy object that spawned three catchers and then deleted itself. I also figured that the chunks could see use elsewhere in different PRs, so I decided on that.

jughu commented:
I can dig 6 tiles of sand make 4 sandstone doors and create a area to have a permament tile protected by ash

Xhuis commented:
Sure you can.
![](https://puu.sh/u9exr/db3ace596c.png)

Mindustry commented:
@Xhuis 
You made it incorrectly. 
X = basalt D = door
XDX
DXD
XDX
I'd go in game to take a picture but I don't have time

Xhuis commented:
Aah, I assumed by "tile" he meant a 2x2 tile. That should work, then, and probably does. I won't cover that in this PR but I'll make another one to patch it out because I'm fairly sure that that isn't intentional.

Mindustry commented:
No, no, no. It's completely intentional and has been in the game for a long time. It's so ghost roles like golem ship and ash walkers can expand their reach and not always be forced to run into a small area whenever an ash storm hits. 

I honestly say no to this PR as this isn't needed. There's already plenty of ways to avoid ash storms or even avoid their damage. This just adds an even easier way to do it for no reason.

Xhuis commented:
...huh. I made this under the assumption that there was no other way. Shows my age, I guess. That would've prevented a good number of my own deaths.

I suppose this has no point, then!

Xhuis commented:
Might re-open this at some point, though.

Xhuis commented:
Well, at least two people on the forums still want this, so I'll keep it as an alternative for people that don't want to have to build four doors every time an ash storm hits.

jughu commented:
Yeah this is much cooler then the doors

jughu commented:
Could you upgrade the ash catcher capicity with watcher restraints

Xhuis commented:
I'd say it's not necessary. The catcher can survive around three storms before it falls apart, and if you're reinforcing a place with a lot of these you'll likely be around more than that.

Doppeltongue commented:
This seems like a thematically apropriate thing for ashwalker lizards to have.

PKPenguin321 commented:
Make the range it protects from storms from like 3x3 or 4x4 and make it non dense so you can walk past it and then it will be worth building over doors

Xhuis commented:
They're non-dense already, don't worry.

PKPenguin321 commented:
Just make them 3x3, trust me it will be fine

Xhuis commented:
Sure, that's nice and all but... I'd like you to look at the sprites in the OP. Now imagine me having to make a 3x3 one.

You see the problem here?

PKPenguin321 commented:
It doesn't need a 3x3 sprite just have a 3x3 range of effect

KorPhaeron commented:
Sorry but I really don't like the thought of cheap and easy mobile storm protection for ash lizards

Xhuis commented:
I guess I could just make them immobile...

PKPenguin321 commented:
They weren't immobile??

Ok well do that then do the 3x3 thing

Xhuis commented:
They couldn't be pulled around but could be taken down and moved elsewhere by hand. Easily removed.

Kromgar commented:
@Xhuis  Just make them immobile and you can't take them down. I really like this idea especially as a miner when you don't want to purchase shelter capsules

Xhuis commented:
I'll do some more work in this once I get home, assuming that @KorPhaeron agrees to reopen it when I address hid concerns. I can't reopen PRs that a maintainer closed.

Xhuis commented:
@KorPhaeron I pushed a commit but it apparently doesn't show up on closed PRs. Mind reopening this? The changes are that they can't be moved once put down and protect a 3x3 area now.
